gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,133 @@
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsTime;
+
+/**
+ * Validates for each trip that the travel speed between each stop is not too fast.
+ *
+ * <p>Generated notice: {@link FastTravelBetweenStopsNotice}.
+ */
+@GtfsValidator
+public class TripTravelSpeedValidator extends FileValidator {
+  private final static float METER_PER_SECOND_TO_KMH_CONVERSION_FACTOR = 3.6f;
+  private final static int FAST_SPEED_METERS_PER_SECOND = 42; // approx 150 km/h
+  private static final int KILOMETER_TO_METER_CONVERSION_FACTOR = 1000;
+
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+
+  @Inject GtfsTripTableContainer tripTable;
+
+  @Inject GtfsStopTableContainer stopTable;
+
+  private static ShapeFactory getShapeFactory() {
+    return GEO.getShapeFactory();
+  }
+
+  private static DistanceCalculator getDistanceCalculator() {
+    return GEO.getDistCalc();
+  }
+
+  public double distanceBetweenMeter(double fromLat, double fromLng, double toLat, double toLng) {
+    final ShapeFactory shapeFactory = getShapeFactory();
+    final DistanceCalculator distanceCalculator = getDistanceCalculator();
+    final Point origin = shapeFactory.pointXY(fromLng, fromLat);
+    final Point destination = shapeFactory.pointXY(toLng, toLat);
+    return (DistanceUtils.DEG_TO_KM * distanceCalculator.distance(origin, destination)
+        * KILOMETER_TO_METER_CONVERSION_FACTOR);
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (GtfsTrip trip : tripTable.getEntities()) {
+      final String tripId = trip.tripId();
+      List<GtfsStopTime> stopTimesForTrip = stopTimeTable.byTripId(tripId);
+
+      // Save the information about the previous stop to compare
+      GtfsTime prevStopDepartureTime = null;
+      double prevStopLatitude = -91;
+      double prevStopLongitude = -181;
+
+      // used to accumulate distance between stops with same arrival and departure times
+      int prevStopAccumulatedDistanceMeter = 0;
+      final List<Integer> prevStopAccumulatedStopSequence = new ArrayList<>();
+
+      // Iterate through the stops consecutively based on time
+      for (GtfsStopTime stopTime : stopTimesForTrip) {
+        GtfsStop currentStop = stopTable.byStopId(stopTime.stopId());
+        double currentStopLat = currentStop.stopLat();
+        double currentStopLon = currentStop.stopLon();
+
+        GtfsTime currentArrivalTime = stopTime.arrivalTime();
+        double distanceFromPreviousStopMeter = 0;
+
+        boolean sameArrivalAndDeparture = false;
+
+        if (prevStopDepartureTime != null && currentArrivalTime != null) {
+          sameArrivalAndDeparture = currentArrivalTime.equals(prevStopDepartureTime);
+
+          if (!sameArrivalAndDeparture) {
+            int durationSecond = currentArrivalTime.getSecondsSinceMidnight()
+                - prevStopDepartureTime.getSecondsSinceMidnight();
```

Comment:
Are we handling the case this is negative correctly?

Reply:
I am now considering the case currentArrivalTime.isBefore(prevStopDepartureTime) below